src/lib.rs

```diff
@@ -42,6 +42,30 @@ fn update() -> Result<(), Box<::std::error::Error>> {
 
 Run the above example to see `self_update` in action: `cargo run --example github`
 
+Amazon S3 is also supported as the backend to check for new releases. Provided a `bucket_name`
+and `asset_prefix` string, `self_update` will look up all matching files using the following format
+as a convention for the filenames: `<asset name>-<semver>-<platform/target>.<extension>`.
```

Comment:
is `asset name` here the meant to be `asset_prefix`?

Reply:
No, the asset name is expected to be the name of the asset that is being published as released, followed by the `-` with semver , etc., that's how we parse it for figuring out the name and version of an asset.
The asset prefix is something to filter them out when fetching them from S3, it's matched against the filename, and it's used as part of the API URL itself: `https://<bucket name>.s3.amazonaws.com/?list-type=2&max-keys=<N>&prefix=<asset prefix>`